src/reconcile/rec_visibility.c

```diff
@@ -264,20 +279,24 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             continue;
         }
 
+        /* Ignore prepared updates if it is not eviction. */
```

Comment:
is it not better to yield and wait if the state is WT_PREPARE_LOCKED ?

Reply:
`WT_PREPARE_LOCKED` of an update during reconciliation is possible only in the code flow of `checkpoint` as this scenario doesn't occur in `eviction`.  Waiting for updates to let it resolved by the checkpoint may increase its total duration. I agree that are advantages of waiting also as there is no need to rewrite the disk page again until no writes happen on the page.

Any opinions from other reviewers?

Reply:
Discussed with various people and decided to not to yield by the checkpoint for `WT_PREPARE_LOCKED` updates.